pkg/controller/windowsmachineconfig/windowsmachineconfig_controller.go

```diff
@@ -215,7 +270,7 @@ func chooseRandomNode(windowsVMs map[types.WindowsVM]bool) types.WindowsVM {
 }
 
 // removeWorkerNode terminates the underlying VM and removes the given vm from the list of VMs
-func (r *ReconcileWindowsMachineConfig) removeWorkerNode(vm types.WindowsVM) error {
+func (r *ReconcileWindowsMachineConfig) removeWorkerNode(vm types.WindowsVM) *reconcileError {
```

Comment:
Is there a reason to use a pointer for `reconcileError`. The pattern I have typically seen is not using pointers for `error`. @ravisantoshgudimetla what are your thoughts on this?

Reply:
I used a pointer so we can do the normal error checking pattern of if `!= nil`

Reply:
You should still be returning `error` everywhere. `reconcileError` should just be one of the types of errors that can be returned by our methods. You should not force us to use `reconcileError` always.